### PR TITLE
Networking zperf fixes

### PR DESCRIPTION
**Pull request by jukkar**
As there is no real documentation for iPerf2 protocol, it is not sure how much these actually help with the issues. At least I am seeing some reasonable output when using iPerf 2.0.9. The iPerf 2.0.13 did not work as expected with zperf.

### CODE REVIEW

File: samples/net/zperf/README.rst [hunk 1]
```diff
@@ -12,7 +12,14 @@ evaluate network bandwidth.
 Features
 *********
 
-- Compatible with iPerf_2.0.5.
+- Compatible with iPerf_2.0.9. Note that UDP uploader (client) testing did
+  not work properly in iPerf_2.0.13 and error message like this is printed
```
Comment (dbkinder) [inline samples/net/zperf/README.rst]:
```suggestion
  not work properly in iPerf_2.0.13: an error message like this is printed
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: samples/net/zperf/README.rst (hunk 1 above)
Comment (dbkinder) [reply samples/net/zperf/README.rst]: BTW, can you mention if this is a problem with iPerf or with our code (and will be fixed later)?

**Comment by jukkar** [comment]
BTW, the  #14737 is mandatory to have in order to get zperf working properly.

File: samples/net/zperf/README.rst (hunk 1 above)
Comment (jukkar) [reply samples/net/zperf/README.rst]: Impossible to know, the iPerf protocol is not really documented, we are just reverse engineering this from wireshark output. Probably we would need to change zephyr implementation but I will leave this for future.

**Comment by jukkar** [comment]
Updated the readme file according to comments.

**Review by tbursztyka — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
+1 for doc changes

**Pull request merged**